### ISSUE TITLE
Problem with UDFs returning Map?

### ISSUE THREAD

**Issue by joewood** [confluentinc/ksql#525]
I have a UDF function that returns a `Map<String,String>`.  The registration of the function is defined as follows:
```java
KsqlFunction mykudf = new KsqlFunction(SchemaBuilder.map(Schema.STRING_SCHEMA,Schema.STRING_SCHEMA).build(), Arrays.asList(Schema.STRING_SCHEMA), "MyMapFunc", MyKudf.class);
addFunction(mykudf);
 ```

The KUDF class's evalulate function returns a `HashMap<String,String>`:
```java
 public Object evaluate(Object... args) {
    Map<String,String> ret = new HashMap<>();
     // .....
    return ret;
}
```

When used in a KSQL statement I get the following error:
`Code generation failed for SelectValueMapper`.  The stacktrace reports the error as:

```
'Code generation failed for SelectValueMapper\r\nCaused by: Line 1, Column 10: Cannot determine simple type name "HashMap"',
```

What Java type should be used to map to a Map in KSQL?

**Comment by apurvam** [comment]
@joewood isn't this fixed by #2029 ?

**Comment by big-andy-coates** [comment]
Hi @joewood if you switch to using the new UDF annotations returning complex types in now supported. So closing this issue.  Please feel free to reopen if you continue to have issues.

Thanks for raising your issue!
